src/components/Button/Button.tsx

```diff
@@ -0,0 +1,63 @@
+import styled from "styled-components";
+
+type ButtonType = "primary" | "secondary";
+
+export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
+  type?: ButtonType;
+}
+
+export interface StyledButtonProps {
+  styleType: ButtonType;
+}
+
+export const Button = ({ type = "primary", ...delegated }: ButtonProps) => (
+  <StyledButton styleType={type} {...delegated} />
+);
+
+const BaseButton = styled.button<StyledButtonProps>`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+
+  padding-left: ${props => props.theme.click.button.basic.space.x};
```

Comment:
what do you think of turning this into a function and having the theme right at the top:
ex: 
```typescript
styled.button<StyledButtonProps>(({theme}) => `
  padding-left: ${theme.click.button.basic.space.x};
`)
```